lgc/builder/SubgroupBuilder.cpp

```diff
@@ -459,10 +471,12 @@ Value *SubgroupBuilder::CreateSubgroupShuffleDown(Value *const value, Value *con
 //
 // @param groupArithOp : The group arithmetic operation.
 // @param value : An LLVM value.
-// @param clusterSize : The cluster size.
+// @param ctSize : The cluster size.
 // @param instName : Name to give final instruction.
 Value *SubgroupBuilder::CreateSubgroupClusteredReduction(GroupArithOp groupArithOp, Value *const value,
-                                                         Value *const clusterSize, const Twine &instName) {
+                                                         Value *const ctSize, const Twine &instName) {
+  auto waveSize = getInt32(getShaderWaveSize());
+  Value *clusterSize = CreateSelect(CreateICmpUGT(ctSize, waveSize), waveSize, ctSize);
```

Comment:
The goal here is to clamp the clusterSize to however many lanes are really there, so I'm pretty sure you can use subgroupSize instead of waveSize. That's better because `subgroupSize <= waveSize`, so if it's different it gives you a tighter clamp and saves code.

Reply:
You're right, I got confused.